package-res/resources/web/pentaho/visual/base/model.js

```diff
@@ -132,68 +126,13 @@ define([
             return ActionResult.reject(e);
           }
 
-          return this._setAction("selectionFilter", newSelectionFilter); //setting to null assigns the default value
-        },
-
-
-        /**
-         * Executes the will/did/rejected event loop associated with a given action.
-         *
-         * @param {function} coreAction - The action to be executed.
-         * @param {pentaho.visual.base.events.Will} will - The "will:" event object.
-         * @param {function} Did - The constructor of the "did:" event.
-         * @param {function} Rejected - The constructor of the "rejected:" event.
-         * @return {pentaho.lang.ActionResult} The result object.
-         * @protected
-         */
-        _doAction: function(coreAction, will, Did, Rejected){
-          if(this._hasListeners(will.type))
-            this._emitSafe(will);
-
-          var result = will.isCanceled ? ActionResult.reject(will.cancelReason) : coreAction.call(this, will);
-
-          if(result.error) {
-            if(this._hasListeners(Rejected.type)) {
-              this._emitSafe(new Rejected(this, result.error, will));
-            }
-          } else {
-            if(this._hasListeners(Did.type)){
-              this._emitSafe(new Did(this, result.value, will));
-            }
-          }
-          return result;
-        },
-
-        /**
-         * Sets the value of a property and returns a [result]{@link pentaho.lang.ActionResult} object.
-         *
-         * This method is supposed to set the value of a property, trigger event loops and return a result.
-         * Currently, the "did:change:selectionFilter" is always triggered.
-         *
-         * @param {nonEmptyString} property - Name of the property to set.
-         * @param {*} value - Value to assign to the property.
-         * @return {pentaho.lang.ActionResult} The result object.
-         * @private
-         */
-        _setAction: function(property, value){
-          var result;
-          try {
-            this.set(property, value);
-            result = ActionResult.fulfill(value);
-            // Currently this method is only used for selectionFilter
-            if(this._hasListeners(DidChangeSelection.type)) {
-              this._emitSafe(new DidChangeSelection(this, result.value, new WillChangeSelection(this, {})));
-            }
-          } catch(e) {
-            result = ActionResult.reject(e);
-          }
-          return result;
+          return this.set("selectionFilter", newSelectionFilter || new filter.Or()); //setting to null assigns the default value
         },
 
         /**
          * Executes an action when the user interacts with a visual element, normally by double clicking it.
          *
-         * The flow starts by triggering the event {@link pentaho.visual.base.events.WillExecute|will:execute}.
+         * The flow starts by triggering the event {@link pentaho.visual.base.events.WillExecute|"will:execute"}.
```

Comment:
Please see http://usejsdoc.org/tags-event.html and declare the class to trigger the events that it triggers.
Also, it's "emitting", not "triggering".